dist/tools/ci/static_tests.sh

```diff
@@ -61,7 +61,11 @@ if [ -n "${CI_BASE_COMMIT}" ]; then
     git tag "${CI_BASE_BRANCH}" "${CI_BASE_COMMIT}"
 fi
 
-CI_BASE_BRANCH=${CI_BASE_BRANCH:-master}
+if [ -z "${GITHUB_RUN_ID}" ]; then
+    # only default to master when not running in a GitHub action
+    # (so GitHub can check release branches too)
```

Comment:
I don't understand this comment and why this if block is needed. Isn't CI_BASE_BRANCH env passed to docker container anyway (`-e CI_BASE_BRANCH=${{ github.base_ref }}`) ?

Reply:
If remains unset, so does `BASE_BRANCH` and thus changed files will return _all_ files (not just the once changed since `CI_BASE_BRANCH`, which when we _are_ on that branch will be no files), which we want for nightlies and branch checks.